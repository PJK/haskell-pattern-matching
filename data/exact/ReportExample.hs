f :: Bool -> Bool -> Int
f _    True  = 1
f True True  = 2
f _    False = 3
