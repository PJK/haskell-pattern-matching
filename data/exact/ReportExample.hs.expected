{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "f",
                {
                    "tag": "InaccessibleRhs",
                    "contents": [
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "True",
                                []
                            ]
                        },
                        {
                            "tag": "ConstructorPattern",
                            "contents": [
                                "True",
                                []
                            ]
                        }
                    ]
                }
            ]
        ],
        [
            [
                "f",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "False",
                                    []
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "False",
                                    []
                                ]
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
